### ISSUE TITLE
PodOverhead: add metrics for improved monitoring/observability

### ISSUE THREAD

**Issue by egernst** [kubernetes/kubernetes#87259]
**What would you like to be added**:

Improved observability for PodOverhead usage. In particular, it would be good to have metrics in place for the number of pods running which utilize an overhead.

Further, it would be good to breakdown these metrics by pod status.

**Why is this needed**:

Users will then easily be able to determine if pod overhead is being utilized in the system, and relative stability of pods running with and without overhead.

**Comment by alejandrox1** [comment]
@egernst have there been any updates on this? is there a plan in place to begin this work ?

**Comment by egernst** [comment]
Hey @alejandrox1 - I was waiting on my changes until I saw original RuntimeClass metrics working.  See https://github.com/kubernetes/kubernetes/pull/87627

I expect to have this tested and a PR with unit tests in place in the next week.

**Comment by egernst** [comment]
@tallclair @dashpole - I could use a soundboard with this.

I was using the existing RuntimeClass monitoring code, which makes use of the [instrumentedRuntimeService](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/instrumented_services.go#L29). This 'wrapper' is quite nice for being able to gather metrics/instrumentation.

For PodOverhead, it is more challenging to determine if an overhead is being utilized, since this isn't part of the CRI API, or something that is understood by the RuntimeService.

Basically, this is less straight forward and I'm wondering if we gather similar metrics elsewhere in kubelet? Basically, based on a key on a PodSpec value, filter by pod-status.

**Comment by dashpole** [comment]
I am generally opposed to the kubelet omitting metrics about the state of kubernetes objects.  I would suggest trying to add this to kube-state-metrics instead.

**Comment by egernst** [comment]
Through kube-state-metrics/#1053, I think we have in place the appropriate metrics now for providing this visibility. Closing.
